pkg/rbac/ensure/ensurer.go

```diff
@@ -0,0 +1,81 @@
+package ensure
+
+import (
+	"errors"
+	"fmt"
+
+	"github.com/rancher/wrangler/pkg/generic"
+)
+
+// Ensurer is used to ensure that a group of objects has been created in k8s, and that all related corrupted versions
+// of the object are properly processed.
+type Ensurer[T generic.RuntimeMetaObject] struct {
+	// ExistingStrategy is the strategy for retrieving existing objects that should be compared with the requested
+	// objects
+	ExistingStrategy ExistingStrategy[T]
```

Comment:
Maybe 
```suggestion
	GetExistingObjectStrategy GetExistingObjStrategy[T]
```
or
```suggestion
	CurrentObjectGetter CurrentObjGetter[T]
```
`ExistingStrategy`, especially in conjunction with `ValidStrategy` or `InvalidStrategy`, makes me think this is how to process existing objects